cmake/macros/macro_deal_ii_query_git_information.cmake

```diff
@@ -27,30 +28,28 @@
 #       GIT_BRANCH
 #       GIT_REVISION
 #       GIT_SHORTREV
+#       GIT_TAG
 #
-# If this macro is called within the deal.II build system the variables are
-# prefixed with DEAL_II_:
+# The macro can be called with an optional PREFIX argument to prefix the variables
 #
-#       DEAL_II_GIT_BRANCH
-#       DEAL_II_GIT_REVISION
-#       DEAL_II_GIT_SHORTREV
+#       PREFIX_GIT_BRANCH
+#       PREFIX_GIT_REVISION
+#       PREFIX_GIT_SHORTREV
+#       PREFIX_GIT_TAG
 #
 
 MACRO(DEAL_II_QUERY_GIT_INFORMATION)
 
   MESSAGE(STATUS "Query git repository information.")
 
-  #
-  # If DEAL_II_BASE_NAME is defined and DEAL_II_PROJECT_CONFIG_INCLUDED was
-  # not set, we assume that we are called from within the deal.II build
-  # system. In this case we prepend all variables by "DEAL_II_"
-  #
-  IF( DEFINED DEAL_II_BASE_NAME AND
-      NOT DEFINED DEAL_II_PROJECT_CONFIG_INCLUDED )
-    SET(_prefix "DEAL_II_")
-  ELSE()
-    SET(_prefix "")
-  ENDIF()
+  SET(_prefix "")
+  # Set prefix if provided
+  LIST(LENGTH ${ARGN} _num_extra_args)
+  IF (${_num_extra_args} GREATER 0)
+    LIST(GET ${ARGN} 0 _prefix)
+    SET(_prefix "${_prefix}_")
+    MESSAGE(STATUS "Store results with a prefix " ${_prefix})
+  ENDIF ()
```

Comment:
What about replacing this with a simple
```
IF(NOT "${ARGN}" STREQUAL "")
  SET(_prefix "${ARGN}_")
ENDIF()
```

Further, please no call to `MESSAGE`.